### PR TITLE
Update IWSLT18 (ST-TED) ASR recipe

### PR DESCRIPTION
**Pull request by hirofumi0810**
- Changed to Transformer
- Added SpecAugment
- Same refactoring as #2901 
- Added patch for mwerSegmenter, which is written by `python2`

### CODE REVIEW

**Comment by hirofumi0810** [comment]
@sw005320 This PR is ready.

File: utils/average_checkpoints.py [hunk 1]
```diff
@@ -115,7 +121,13 @@ def get_parser():
         default="",
         type=str,
         nargs="?",
-        choices=["acc", "bleu", "cer_ctc", "loss", "perplexity"],
+        choices=["acc", "bleu", "cer", "cer_ctc", "loss", "perplexity"],
+    )
+    parser.add_argument(
```
Comment (sw005320) [inline utils/average_checkpoints.py:126]: What are the intentions of this option?

Comment (hirofumi0810) [reply utils/average_checkpoints.py:126]: @sw005320 Sorry. In the initial experiments, I found `cer_ctc` option gave better performance on this corpus. However, it turned out that `acc` option is better than it after parameter tuning. Although `cet_ctc` does not give the best performance, I just want to keep it as an option.

Comment (sw005320) [reply utils/average_checkpoints.py:126]: OK!

File: utils/average_checkpoints.py [hunk 2]
```diff
@@ -115,7 +121,13 @@ def get_parser():
...
+        "--max-epoch",
+        default=10000000,
+        type=int,
+        nargs="?",
```
Comment (sw005320) [inline utils/average_checkpoints.py:130]: How about this option?
How is this option used?

Comment (hirofumi0810) [reply utils/average_checkpoints.py:130]: @sw005320 I usually use this option to check overfitting and compare multiple models epoch by epoch. If you don't prefer it, I'll remove it. But I'd like to add it in another PR.

Comment (sw005320) [reply utils/average_checkpoints.py:130]: Thanks!
We can keep this option then.

**Comment by sw005320** [comment]
I just merged!
Thanks, @hirofumi0810!

**Pull request merged**